.github/workflows/preview.yml

```diff
@@ -0,0 +1,42 @@
+name: Amplify Preview
+
+on:
+  pull_request:
+    types: [labeled]
+
+jobs:
+  deploy:
+    if: contains(github.event.pull_request.labels.*.name, 'Preview Environment') && github.event.action == 'labeled'
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    permissions:
+      id-token: write
+      contents: write
+      pull-requests: write
+      repository-projects: write
```

Comment:
Are all of these permissions needed? I would think it would only need `id-token` like the other workflow.

Reply:
Gotcha. Even `repository-projects`?

Reply:
According to what I've read, yes. However, we can test it with a few more commits, but it will take the rest of the morning at least to do so.

Reply:
According to [this doc](https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28#create-an-issue-comment) we may be able to get away with just `pull-requests: write`